File: python-client-sdk/src/orbapi/models/shared/account.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from orbapi import utils
from ..shared import address as shared_address


@dataclass_json
@dataclasses.dataclass
class AccountAssociations:
    contact_ids: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contactIds') }})
    deal_ids: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('dealIds') }})
    event_ids: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('eventIds') }})
    lead_ids: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('leadIds') }})
    note_ids: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('noteIds') }})
    owner_user_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ownerUserId') }})
    task_ids: list[str] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('taskIds') }})
    

@dataclass_json
@dataclasses.dataclass
class Account:
    r"""Account
    (Alias: company, organization) An organization involved with your business.
    """
    
    associations: AccountAssociations = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('associations') }})
    created_time: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('createdTime') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    modified_time: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('modifiedTime') }})
    native_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('nativeId') }})
    additional: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('additional') }})
    address: Optional[shared_address.Address] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('address') }})
    annual_revenue: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('annualRevenue') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    industry: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('industry') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    number_of_employees: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('numberOfEmployees') }})
    phone: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('phone') }})
    website: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('website') }})
    

File: go-client-sdk/pkg/models/shared/account.go
package shared

type AccountAssociations struct {
	ContactIds  []string `json:"contactIds"`
	DealIds     []string `json:"dealIds"`
	EventIds    []string `json:"eventIds"`
	LeadIds     []string `json:"leadIds"`
	NoteIds     []string `json:"noteIds"`
	OwnerUserID string   `json:"ownerUserId"`
	TaskIds     []string `json:"taskIds"`
}

// Account
// (Alias: company, organization) An organization involved with your business.
type Account struct {
	Additional        map[string]interface{} `json:"additional,omitempty"`
	Address           *Address               `json:"address,omitempty"`
	AnnualRevenue     *float64               `json:"annualRevenue,omitempty"`
	Associations      AccountAssociations    `json:"associations"`
	CreatedTime       string                 `json:"createdTime"`
	Description       *string                `json:"description,omitempty"`
	ID                string                 `json:"id"`
	Industry          *string                `json:"industry,omitempty"`
	ModifiedTime      string                 `json:"modifiedTime"`
	Name              *string                `json:"name,omitempty"`
	NativeID          string                 `json:"nativeId"`
	NumberOfEmployees *float64               `json:"numberOfEmployees,omitempty"`
	Phone             *string                `json:"phone,omitempty"`
	Website           *string                `json:"website,omitempty"`
}
